velox/common/file/File.h

```diff
@@ -260,4 +235,66 @@ class LocalWriteFile final : public WriteFile {
   mutable long size_;
 };
 
+// An abstract FileSystem
```

Comment:
this looks great, but any chance we can split this PR and submit the changes to this File interface first, then another one adding the s3 stuff on top of it?

Reply:
Done https://github.com/facebookincubator/velox/pull/293